web/src/app/components/HighlightText.tsx

```diff
@@ -36,6 +38,12 @@ export default function HighlightText({originalText, userInput, onChange, onCorr
         setGameText(gameData)
     }, [])
 
+    useEffect(() => {
```

Comment:
this doesn't need to be an useEffect, since hasGameStarted will get updated on each render anyway

Reply:
It won't get updated after it hits 0